src/main/scala/Models.scala

```diff
@@ -1,21 +1,38 @@
+import spray.json._
+import DefaultJsonProtocol._
+import sangria.execution.deferred.HasId
+import sangria.schema._
+import sangria.macros.derive._
+
 case class Category(
   id: Int,
-  name: String
-) {
-  def styles(allStyles: List[Style]): List[Style] =
-    allStyles.filter(_.category_id == id)
+  name: String)
+
+object Category {
+  implicit val categoryHasId = HasId[Category, Int](_.id)
```

Comment:
I see [`HasId` lives here](https://github.com/sangria-graphql/sangria/blob/663f654647dfc5ab84dc9cea7274d9493bb269c9/src/main/scala/sangria/execution/deferred/HasId.scala), do you mind explaining what it's for?